restApi.py

```diff
@@ -138,24 +139,77 @@ def delete_booked_rooms(BookingId):
    RoomIdList = [item[0] for item in c.fetchall()]
    if(RoomIdList.__len__() > 0 ):
       for RoomId in RoomIdList:
          RoomIdInt = int(RoomId)
+         print('RoomIdInt:')
+         print(RoomIdInt)
+
          c.execute("SELECT Name FROM ROOMS WHERE id='%s'" % RoomIdInt)
          roomNameList = [item[0] for item in c.fetchall()]
          roomName = ''.join(roomNameList)
-         c.execute("SELECT UserId FROM BOOKINGS WHERE BookingId='%s'" % BookingId)
-         UserIdList = [item[0] for item in c.fetchall()]
-   if(UserIdList.__len__() > 0 ):
-      for UserId in UserIdList:
-         UserIdInt = int(UserId)
-         c.execute("SELECT Email FROM USERS WHERE id='%s'" % UserIdInt)
+         print(roomName)
+
+         c.execute("SELECT UserId FROM WATCHED WHERE RoomId='%s'" % RoomIdInt)
+         WatchedUserIdList = [item[0] for item in c.fetchall()]
+         for WatchedUserId in WatchedUserIdList:
+            WatchedUserIdInt = int(WatchedUserId)
+            print('WatchedUserIdInt:')
+            print(WatchedUserIdInt)
+
+            c.execute("SELECT StartTime FROM WATCHED WHERE RoomId='%s' AND UserId='%s'" % (RoomIdInt, WatchedUserIdInt))
+            WatchedStartTimeList = [item[0] for item in c.fetchall()]
+            for WatchedStartTimeString in WatchedStartTimeList:
+               print(WatchedStartTimeString)
+               f = '%Y-%m-%d %H:%M:%S'
+               WatchedStartTime = datetime.datetime.strptime(WatchedStartTimeString, f)
+               
+            c.execute("SELECT EndTime FROM WATCHED WHERE RoomId='%s' AND UserId='%s'" % (RoomIdInt, WatchedUserIdInt))
+            WatchedEndTimeList = [item[0] for item in c.fetchall()]
+            for WatchedEndTimeString in WatchedEndTimeList:
+               print(WatchedEndTimeString)
+               f = '%Y-%m-%d %H:%M:%S'
+               WatchedEndTime = datetime.datetime.strptime(WatchedEndTimeString, f)
+            c.execute("SELECT UserId FROM BOOKINGS WHERE BookingId='%s'" % BookingId)
+            BookingUserIdList = [item[0] for item in c.fetchall()]
+
+   if(BookingUserIdList.__len__() > 0 ):
+      for BookingUserId in BookingUserIdList:
+         BookingUserIdInt = int(BookingUserId)
+         c.execute("SELECT Email FROM USERS WHERE id='%s'" % BookingUserIdInt)
+         emailList = [item[0] for item in c.fetchall()]
+         email = ''.join(emailList)
+         sendEmail(email, roomName, 'booking')
+      else:
+         print("Bookings table has no more results")
+   if(WatchedUserIdList.__len__() > 0 ):
+      for WatchedUserId in WatchedUserIdList:
+         WatchedUserIdInt = int(WatchedUserId)
+         c.execute("SELECT Email FROM USERS WHERE id='%s'" % WatchedUserIdInt)
          emailList = [item[0] for item in c.fetchall()]
          email = ''.join(emailList)
-         sendEmail(email, roomName)
+         if BookingStartTime == WatchedStartTime and BookingEndTime == WatchedEndTime:
```

Comment:
In this situation, if the room had previously been booked for 12:00-13:00 and you were only watching for the duration 12:30-13:00 then you wouldn't get a notification as the start time did not match.

I wrote a similar comparison for the `/watch-list/<user>` and `/booked-rooms` endpoints:
```c.execute("SELECT COUNT(*) from BOOKINGS WHERE BOOKINGS.RoomId=%s AND ((BOOKINGS.StartTime BETWEEN '%s' AND '%s') OR (BOOKINGS.EndTime BETWEEN '%s' AND '%s') OR ('%s' BETWEEN BOOKINGS.StartTime AND BOOKINGS.EndTime))" % (int(item[2]), str(item[4]), str(item[5]), str(item[4]), str(item[5]), str(item[4])))```
Maybe do a check against the booking table using a similar SQL query as above to check if the room has actually now become available for your desired slot and if so send the email.

Reply:
Does the between also include the end time eg if I write BETWEEN 13:00 and 14:00  does it include 14:00 ?

Reply:
That's a good point, maybe using a combination of >,<,<=,>= etc would be better. That is something I might have to change in my earlier usage...